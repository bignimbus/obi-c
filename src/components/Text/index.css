.text {
  color: var(--theme-contrast-color);
  font-family: 'IBM Plex Sans', sans-serif;
  line-height: 1.35;
}

.text--inverted {
  color: var(--theme-color);
}

.text--inverted-darker {
  color: var(--theme-color--darker);
}

.text--xs {
  font-size: 12px;
}

.text--sm {
  font-size: 16px;
}

.text--md {
  font-size: 20px;
}

.text--lg {
  font-size: 24px;
}

.text--xl {
  font-size: 28px;
}
