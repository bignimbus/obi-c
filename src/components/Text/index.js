import './index.css';
import React from 'react';
import PropTypes from 'prop-types';

const Text = ({
  children,
  inverted,
  size = 'sm',
  invertedDark,
}) => (
  <span
    className={`
      text
      text--${size}
      ${inverted ? 'text--inverted' : ''}
      ${invertedDark ? 'text--inverted-dark' : ''}
    `}
  >
    { children }
  </span>
);

Text.propTypes = {
  size: PropTypes.string,
  inverted: PropTypes.bool,
  invertedDark: PropTypes.bool,
  children: PropTypes.node.isRequired,
};

export default Text;
