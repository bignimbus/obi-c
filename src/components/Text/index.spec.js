import React from 'react';
import renderer from 'react-test-renderer';
import Text from '.';

const renderComponent = props => renderer.create(
  <Text {...props}>
    Lorem ipsum
  </Text>,
);

describe('Text', () => {
  test('default', () => {
    const element = renderComponent();
    expect(element.toJSON()).toMatchSnapshot();
  });

  test('with specified size prop', () => {
    const element = renderComponent({ size: 'md' });
    expect(element.toJSON()).toMatchSnapshot();
  });

  test('inverted', () => {
    const element = renderComponent({ inverted: true });
    expect(element.toJSON()).toMatchSnapshot();
  });

  test('inverted-darker', () => {
    const element = renderComponent({ invertedDark: true });
    expect(element.toJSON()).toMatchSnapshot();
  });
});
