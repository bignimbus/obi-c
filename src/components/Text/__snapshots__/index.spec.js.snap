// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text default 1`] = `
<span
  className="
      text
      text--sm
      
      
    "
>
  Lorem ipsum
</span>
`;

exports[`Text inverted 1`] = `
<span
  className="
      text
      text--sm
      text--inverted
      
    "
>
  Lorem ipsum
</span>
`;

exports[`Text inverted-darker 1`] = `
<span
  className="
      text
      text--sm
      
      text--inverted-dark
    "
>
  Lorem ipsum
</span>
`;

exports[`Text with specified size prop 1`] = `
<span
  className="
      text
      text--md
      
      
    "
>
  Lorem ipsum
</span>
`;
